eng/pipelines/templates/stages/archetype-cpp-release.yml

```diff
@@ -86,53 +86,69 @@ stages:
                     steps:
                       - checkout: self
 
+                      - pwsh: |
+                          $branchName = "azure-sdk-for-cpp-${{ artifact.Name }}-$(Build.BuildId)"
+                          if ('$(VcpkgForkBranchName)') { 
+                            Write-Host "Using queue time branch name" 
+                            $branchName = '$(VcpkgForkBranchName)'
+                          } 
+                          Write-Host "##vso[task.setvariable variable=PrBranchName]$branchName"
+                        displayName: Set fork branch name
+
                       - task: Powershell@2
                         inputs:
                           filePath: eng/scripts/Initialize-VcpkgRelease.ps1
                           arguments: >-
-                            -GitHubPat $(azuresdk-github-pat)
                             -GitHubRepo "$(Build.Repository.Name)"
                             -PackageSpecPath $(Pipeline.Workspace)/packages/${{artifact.Name}}/package-info.json
                             -SourceDirectory $(Pipeline.Workspace)/packages/${{artifact.Name}}/vcpkg
                             -Verbose
                           pwsh: true
                         displayName: Initialize vcpkg release
 
+                      # There are potential race conditions if this script runs
+                      # in parallel against the same branch name. Release only 
+                      # one package at a time.
                       - pwsh: |
```

Comment:
Minor but I'd suggest splitting the cloning and the copying into different steps.

Reply:
Will that help reduce the chances of a race condition?

Reply:
No that doesn't affect the race conditions. There technically isn't any race condition there but there is a risk that multiple release pipelines hit a merge conflict when they run if they are all targeting the same branch. I think the biggest risk is if you release the same package multiple times targeting the shared branch as that will cause a merge conflict and likely fail.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/1356